From 2fe255e6d9e1c078ffeee4171d2c57a21d4c2ea8 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 15 Jun 2012 15:31:51 +0000
Subject: [PATCH 0289/1076] tools.lib; V1.0.1.10 Mantis  0023249: GPP: support
 two different paper types for a single printer bugfix: resize from size 0 to
 an other size does not reserve memory

git-svn-id: svn://localhost/SelfServiceCommon/trunk@347 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Tools/inc/local_def_msystools.h            | 8 ++++++--
 SelfServiceCommon/Massai/cpp/Tools/src/stringbuffer.cpp   | 7 +++++--
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h b/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
index 163791cd..25deb7c5 100644
--- a/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
+++ b/SelfServiceCommon/Massai/cpp/Tools/inc/local_def_msystools.h
@@ -1,9 +1,13 @@
-#define FILE_VER_BIN     1,0,1,9
-#define FILE_VER_STR     "FileVersion", "1.0.1.9\0"
+#define FILE_VER_BIN     1,0,1,10
+#define FILE_VER_STR     "FileVersion", "1.0.1.10\0"
 
 /*! \page tools.lib tools.lib - Release Notes
  *
  *
+ *  \date 15.06.2012 \li V1.0.1.10
+ *                   \li SteT -  Mantis  0023249: GPP: support two different papertypes for a single printer
+ *                   \li bugfix: resize from size 0 to an other size does not reserve memory
+ *
  *  \date 19.03.2012 \li V1.0.1.9
  *                   \li SteT -  Mantis 0017128: TrcServ: replace communication layer
  *                       added TraceSetLevel and TraceSetLevelDelegate to filter trace by tracelevel/topic
diff --git a/SelfServiceCommon/Massai/cpp/Tools/src/stringbuffer.cpp b/SelfServiceCommon/Massai/cpp/Tools/src/stringbuffer.cpp
index 26505e02..c61f808f 100644
--- a/SelfServiceCommon/Massai/cpp/Tools/src/stringbuffer.cpp
+++ b/SelfServiceCommon/Massai/cpp/Tools/src/stringbuffer.cpp
@@ -137,7 +137,7 @@ namespace MASSAI
       {
         m_capacity= capacity;
       }
-      if(m_size == 0)
+      if(m_capacity == 0)
       {
         m_pBuf= 0;
       }
@@ -146,7 +146,10 @@ namespace MASSAI
         m_pBuf= new char[m_capacity];
         if(pBuf != 0)
         {
-          memcpy(m_pBuf, pBuf, m_size);
+          if(m_size > 0)
+          {
+            memcpy(m_pBuf, pBuf, m_size);
+          }
           if(m_capacity > m_size)
           {
             memset(m_pBuf + m_size, 0, m_capacity - m_size);
-- 
2.41.0.windows.1

